### ISSUE TITLE
Simultaneous DB State Change + Schema Filtering Error

### ISSUE THREAD

**Issue by asummers** [djrobstep/migra#145]
This is technically an issue in `schemainspector` but manifests itself via this project so I thought to open the issue here and ask for guidance. 

My use case is that I have databases with many schemas who migrate out of band. To clean up the `migra` output a bit by organizing by schema, I iterate over the schemas out of band and call `migra` with the `--schema` flag. What I am finding is that when new tables/partitions/schemas/indices/etc. get created while `migra` is running, `schemainspector` blows up when looking at the new additions. Fair enough, this is somewhat reasonable as the DB state rug is being ripped out from underneath of it. But, it seems to be blowing up when given DBs with schemas `abc` and `def`, and running `migra $DB1 $DB2 --schema abc`, simultaneous changes to `def` cause the command to blow up. I would expect `--schema` filtering to cause this to not trip anything (which may come with slight perf benefits??), and/or the commands tolerate the simultaneous changes to the table state better altogether.

I do not currently have stack trace output or a reproduction case, but I can work on getting that. I am seeing this pretty consistently in my setup depending on getting unlucky in the timing of the runs, due to the number of schemas we are comparing.

**Comment by djrobstep** [comment]
Hi, thanks for filing this issue.

Do you mean that new tables are added *during* an inspection or *between* inspections? `schemainspect` should never fail during an inspection because of external changes, because it runs within a transaction with a consistent view of the database state for the duration.

Note that the way the `--schema` flag works is pretty primitive. `schemainspect` always inspects the entire database no matter the presence of that flag. The flag just filters the inspected objects afterwards and removes things not belonging to the specified schema.

Hopefully that clarifies your situation somewhat.

**Comment by asummers** [comment]
Hi. I believe it is being modified _during_, but I'm not completely positive yet. I'm getting key lookup errors from some map from schemas that create a lot of partitions on new partitions, which leads me to suspect it's during the migration. 

Re: the schema filter, that makes sense. If that was able to be moved into `schemainspect` s (apologies on goofing the name originally) omewhat, and only considered tables in the information catalog from that specific schema, that could be a nice optimization.

**Comment by asummers** [comment]
I think I understand how we might be bypassing the expected transaction behavior. What we found is that `migra` was hanging onto idle transactions for minutes at a time, so we added code [here](https://github.com/djrobstep/migra/blob/master/migra/command.py#L19) to close the transaction after read by adding the `isolation_level="AUTOCOMMIT"` opt per the discussion of [this](https://www.oddbird.net/2014/06/14/sqlalchemy-postgres-autocommit/) blogpost. While that's obviously not completely supported behavior, it was bypassing `migra` holding transactions open for 5+ minutes in our case (before the DB killed it as an idle transaction per our PG config).
